17/umbraco-cms/customizing/foundation/repositories/README.md

```diff
@@ -0,0 +1,116 @@
+# Repositories
+Repositories provide a structured way to manage data operations in the Backoffice. They abstract the data access layer, allowing for easier reuse and scalability.
+
+Repositories create separation between domain logic and data access. By providing a known interface for data requests, we can reuse UI components across different domains. For example, we have a generic UX flow for deleting an entity. By supplying this flow with a repository that has a known interface for deletion, we can use the same UX flow to delete any entity. The same applies to Trees, Collections, Workspaces, and more.
+
+Additionally repositories can utilize different data sources depending on the application's state. These sources may include:
+
+* A REST API
+* Offline storage
+* A local cache
+* Etc.
+
+This abstraction ensures that consumers don’t need to worry about how to access data. The repository serves as the Backoffice’s entry point for requesting new data. As a result, we achieve a loosely coupled connection between consumers and data storage procedures, effectively hiding complex implementations.
+
+* **Repository:** defines what data operations are available (get, add, update, delete).
+* **Data Source:** defines how data is fetched or stored.
+
+### Data flow with a repository <a href="#data-flow-with-a-repository" id="data-flow-with-a-repository"></a>
+
+A repository must be instantiated where it is used. It should take an [UmbController](../../umbraco-controller/README.md) as part of the constructor. This ensures that any contexts consumed in the repository are scoped correctly.
+
+A repository can be initialized directly from an element, but will often be instantiated in a [context](../../context-api/README.md), like the Workspace Context.
+
+The data flow when using a repository can be illustrated as follows:
+
+```text
+(Data Source) -> Repository -> (Controller/Context) -> Element
+```
+
+### Using an existing Repository <a href="#using-a-repository" id="using-a-repository"></a>
+
+Often, you will find that data is already available and observable in a [context](./contexts/README.md). In that case, subscribing to the context [state](./states.md) will be the right approach to take. This way, you will receive all runtime updates that occur to the data throughout the session.
+
+If the needed data isn’t available in a context, use a repository to request the data. This will give you that correct data no matter the current application state.
```

Comment:
```suggestion
If the needed data isn’t available in a context, use a repository to request the data. This will give you the correct data no matter the current application state.
```